### ISSUE TITLE
model.eval returns no mask pixels found

### ISSUE THREAD

**Issue by TarekMaylaa** [MouseLand/cellpose#375]
Hello , 

After training the cellpose model on my dataset, the model.eval  returns: "no mask pixel found" with a 0's pixels black image mask.           I don't seem to have this problem when using a pre-trained model for prediction. 

NB: The dataset is formed of 1024x1024 grayscale TIFF images

 Thank you for your assistance. 

Please find attached snippets from my code:

`model = models.CellposeModel(diam_mean = 9.6,
                             pretrained_model= None,
                                 gpu=True,
                                 torch = True
                            )`
`train_dir = "C:/Users/MAYLAA/Desktop/PhD Experiments and Analysis/CellPose/data_cellpose_stardist_test_training/train/models/models/"

cpmodel_path = model.train(X, Y, train_files=train_fileNames, 
                               test_data=x, test_labels=y, test_files=test_fileNames,
                               channels=[0,0], save_path=train_dir, n_epochs=10)

print('>>>> model trained and saved to %s'%cpmodel_path)`

`import cv2

img_path = "C:/Users/MAYLAA/Desktop/PhD Experiments and Analysis/CellPose/data_cellpose_stardist_test_training/testdata/"

img_filename ="'images/tile7 1.tif"

mask_filename = "masks/tile7 1_mask.tif"
img = cv2.imread(img_path+img_filename)

mask = cv2.imread(img_path+mask_filename)

print('"img shape", img.shape)
img = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
plt.imshow(img)`
`fig = plt.figure(figsize=(24,10))

#import cv2
#img_path = 'C:/Users/MAYLAA/Desktop/PhD Experiments and Analysis/CellPose/data_cellpose_stardist_test_training/testdata/images/'
save_path = 'C:/Users/MAYLAA/Desktop/PhD Experiments and Analysis/CellPose/data_cellpose_stardist_test_training/testdata/test_re/'
#img_filename = 'tile7 1.tif'
#img = cv2.imread(img_path+img_filename)
#print('img shape',img.shape)
#img = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)

print('img shape',img.shape)

masks, flows, diams = model.eval(img, diameter= None,channels=[0,0])
img_res = np.array(flows[0])
#img_res =cv2.cvtColor(img_res,cv2.COLOR_BGR2RGB)
res,thresh = cv2.threshold(img_res, 127, 255, cv2.THRESH_BINARY)
plt.subplot(1,2,1)
plt.imshow(masks)
plt.subplot(1,2,2)
plt.imshow(thresh)
print(np.unique(masks))
print('helo',masks.shape)
print(thresh.shape)
cv2.imwrite(save_path+img_filename,img_res)`
`from cellpose import plot

fig = plt.figure(figsize=(12,5))
plot.show_segmentation(fig, img, masks, flows[2], channels=[0,0])
plt.tight_layout()
plt.show()`

**Comment by kevinjohncutler** [comment]
It's possible that training via jupyter notebook is still broken, I remember having issues with that a long time ago. To debug, could you try training with the CLI command and see how that model does?

**Comment by TarekMaylaa** [comment]
@kevinjohncutler Thank you for your response. I tried the training and evaluation process with the CLI commands. During the evaluation, it says that the model path (my trained model) does not exist and perform the prediction with the cyto pre-trained model. Plus, when I am training with a bigger dataset (28 images of 1024x1024) I have a train and validation loss = nan. Do you have any recommendation about  this issue? Thank you

**Comment by carsen-stringer** [comment]
that training set is likely far too small. try downloading the cellpose training set (https://www.cellpose.org/dataset) and adding these images to the training set
